### PR TITLE
Add log level and flag operations

### PR DESCRIPTION
**Pull request by derekbit**
#### Which issue(s) this PR fixes:

Issue Longhorn/longhorn#7939

### CODE REVIEW

File: pkg/spdk/client/basic.go [hunk 1]
```diff
@@ -944,3 +945,88 @@ func (c *Client) NvmfSubsystemGetListeners(nqn, tgtName string) (listenerList []
 
 	return listenerList, json.Unmarshal(cmdOutput, &listenerList)
 }
+
+func (c *Client) LogSetFlag(flag string) (result bool, err error) {
```
Comment (innobead) [inline pkg/spdk/client/basic.go:949]: Let's make sure all exported entities (ex: method, function, etc.) to have a doc comment, even though the function is straight forward.

**Review by innobead — CHANGES_REQUESTED** [review]
Just one comment, others LGTM. 

What's difference among level, print-leve and flag?

**Comment by derekbit** [comment]
> Just one comment, others LGTM.
> 
> What's difference among level, print-leve and flag?

print-level is for stderr/stdout, and level is for syslog.
https://github.com/spdk/spdk/blob/master/lib/log/log.c#L158-L173

File: pkg/spdk/client/basic.go (hunk 1 above)
Comment (derekbit) [reply pkg/spdk/client/basic.go:926]: Added. Thank you.

File: pkg/spdk/client/basic.go [hunk 2]
```diff
@@ -918,3 +919,103 @@ func (c *Client) NvmfSubsystemGetListeners(nqn, tgtName string) (listenerList []
...
+	if err != nil {
+		return nil, err
+	}
+
+	return flags, json.Unmarshal(cmdOutput, &flags)
+}
+
+// LogSetLevel sets the log level.
+//
+// "level": Required. Log print level to set. Supported values are "disabled", "error", "warn", "notice", "info", "debug". Default is "notice".
```
Comment (innobead) [inline pkg/spdk/client/basic.go:969]:
```suggestion
// "level": Required. Supported values are "disabled", "error", "warn", "notice", "info", "debug". Default is "notice".
```

File: pkg/spdk/client/basic.go [hunk 3]
```diff
@@ -918,3 +919,103 @@ func (c *Client) NvmfSubsystemGetListeners(nqn, tgtName string) (listenerList []
...
+	if err != nil {
+		return "", err
+	}
+
+	return strings.Trim(string(level), "\"\n"), nil
+}
+
+// LogSetPrintLevel sets the log print level. The log print level is the level at which log messages are printed to the console.
+//
+// "level": Required. Log print level to set. Supported values are "disabled", "error", "warn", "notice", "info", "debug". Default is "notice".
```
Comment (innobead) [inline pkg/spdk/client/basic.go:997]:
```suggestion
// "level": Required. Supported values are "disabled", "error", "warn", "notice", "info", "debug". Default is "notice".
```

**Review by innobead — APPROVED** [review]

**Pull request merged**